app/models/mapping.rb

```diff
@@ -15,7 +15,35 @@ class Mapping < ApplicationRecord
   validates :name, presence: true
 
   # The possible status of a mapping
-  # 1. "uploaded" This means there's a specification uploaded but not terms mapped
-  # 2. "mapped" It means the terms are confirmed as mapped to the spine
-  enum status: %i[uploaded mapped]
+  # 1. "uploaded" It means that there's a specification uploaded but not
+  #    terms mapped
+  # 2. "in-progress" It means that the user is already mapping terms but
+  #    not yet finished mapping
+  # 3. "mapped" It means the terms are confirmed as mapped to the spine
+  enum status: %i[uploaded in_progress mapped]
```

Comment:
Let's set up the enum with explicit values, like this: 

```ruby
enum status: { uploaded: 0, in_progress: 1, mapped: 2 }
```

That way the statuses won't be accidentally messed up if a new value is added in the wrong place.